qemu/tests/block_iothread_test.py

```diff
@@ -0,0 +1,152 @@
+            num = int(attr[1]) if attr[1] else 0
+            expect_iothreads = set(attr[2].split(",") if attr[2] else [])
+            logger.debug("Expected %s iothread :%s %s", name, num,
+                         expect_iothreads)
+            parent_bus = vm.monitor.qom_get(name, "parent_bus")
+            parent_bus_type = vm.monitor.qom_get(parent_bus, "type")
+            check_name = name
+            if parent_bus_type == "SCSI":
+                check_name = parent_bus.split("/")[3]
+                logger.debug("Ready to check SCSI %s iothread", check_name)
+
+            cmd_out = vm.monitor.qom_get(check_name, iothread_type)
+            logger.debug("Get %s %s out:%s", check_name, iothread_type, cmd_out)
+            if iothread_type == "iothread":
+                cmd_out = cmd_out.replace("/objects/", "").split()
+            else:
+                # iothread-vq-mapping
+                cmd_out = [i["iothread"] for i in cmd_out]
+
+            real_iothreads = set(cmd_out)
+            logger.debug("Real iothread %s :%s", name, real_iothreads)
+
+            if expect_iothreads:
+                if real_iothreads != expect_iothreads:
+                    test.fail("Get unexpeced %s iothread list:%s %s" % (
+                        name, expect_iothreads, real_iothreads))
+            else:
+                if len(real_iothreads) != num:
+                    test.fail("Get unexpeced %s iothread len:%s %s" % (
+                        name, num, len(real_iothreads)))
+
+    def hotplug_disks_test():
+        plug = BlockDevicesPlug(vm)
+        for img in test_images:
+            plug.unplug_devs_serial(img)
+            plug.hotplug_devs_serial(img)
+            _check_disk_in_guest(img)
+
+        check_image_iothread()
+
+    logger = test.log
+
+    vm = None
+    session = None
+    expect_to_fail = params.get("expect_to_fail", "no")
+    err_msg = params.get("err_msg", "undefined unknown error")
+    start_vm = params.get("start_vm")
+    try:
+
+        timeout = params.get_numeric("login_timeout", 180)
+        test_images = params["test_images"].split()
+        host_cmd = params.get("host_cmd")
+        guest_cmd = params.get("guest_cmd")
+        guest_operation = params.get("guest_operation")
+
+        if start_vm == "no":
+            params["start_vm"] = 'yes'
+            env_process.preprocess_vm(test, params, env, params.get("main_vm"))
+
+        error_context.context("Get the main VM", test.log.info)
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+
+        session = vm.wait_for_login(timeout=timeout)
+        check_image_iothread()
+        locals_var = locals()
+        if guest_operation:
+            logger.debug("Execute guest operation %s" % guest_operation)
+            locals_var[guest_operation]()
+
+        logger.debug("Destroy VM...")
+        session.close()
+        vm.destroy()
+        vm = None
+    except (virt_vm.VMCreateError, ShellCmdError) as e:
```

Comment:
Suggest adding virt_vm.VMStartError as well, we may hit this issue. Thanks.
(virt_vm.VMCreateError, virt_vm.VMStartError, ShellCmdError)